OracleWebLogic/samples/12212-msiserver/container-scripts/launcher.sh

```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/bash
```

Comment:
Hi Avi,

Following up on your question about behavior of docker stop with WLS, I experimented and found the following.

**About docker stop**
1. docker stop sends a SIGTERM to the main process initially and then sends a SIGKILL after some time. This is a configurable value, defaults to 10 seconds.
2. WebLogic java process handles SIGTERM to perform a forceShutdown() of the server, cleaning up a few things.
3. When we launch WebLogic using generated shell scripts, sending SIGTERM to the pid of shell script doesn’t invoke the hook mentioned in #2
4. However if we find the pid of the java process (JVM that is running the server) and send a SIGTERM to it, shutdown hook mentioned in #2 is invoked
5. Trying to replicate #3 in Docker image is equally ineffective. More specifically we can try and capture the pid of the process for the WLS startup script being launched as a part of docker container, but trapping SIGTERM (on docker stop) and forwarding to that captured pid is ineffective (like #3)
5. Perhaps we can replicate #4 in docker image i.e. try to trap SIGTERM and translate to a SIGTERM to all running java processes. But I am having trouble with this because we use oracle slim image in PS2 docker image and “ps” is not available.
6. For WLS MSI docker image, I have been using the generated WLS script called startManagedWebLogic.sh. But I believe that observations in #3 and #5 will be the same if we use startWebLogic.sh for admin server in a regular WLS docker image

**About docker kill**

docker kill results in a SIGKILL signal. There is no way to intercept or trap it. That is also the case when we send a SIGKILL signal to a WLS server started with shell scripts (outside docker)